File: generated-sources/csharp-dotnet2/mojang-api/src/main/CsharpDotNet2/Org/OpenAPITools/Model/SecurityChallenge.cs
using System;
using System.Text;
using System.Collections;
using System.Collections.Generic;
using System.Runtime.Serialization;
using Newtonsoft.Json;

namespace Org.OpenAPITools.Model {

  /// <summary>
  /// Binds a question to an answer id, the answer must respect the answer id.
  /// </summary>
  [DataContract]
  public class SecurityChallenge {
    /// <summary>
    /// Gets or Sets Question
    /// </summary>
    [DataMember(Name="question", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "question")]
    public SecurityQuestion Question { get; set; }

    /// <summary>
    /// Gets or Sets Answer
    /// </summary>
    [DataMember(Name="answer", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "answer")]
    public SecurityAnswerId Answer { get; set; }


    /// <summary>
    /// Get the string presentation of the object
    /// </summary>
    /// <returns>String presentation of the object</returns>
    public override string ToString()  {
      var sb = new StringBuilder();
      sb.Append("class SecurityChallenge {\n");
      sb.Append("  Question: ").Append(Question).Append("\n");
      sb.Append("  Answer: ").Append(Answer).Append("\n");
      sb.Append("}\n");
      return sb.ToString();
    }

    /// <summary>
    /// Get the JSON string presentation of the object
    /// </summary>
    /// <returns>JSON string presentation of the object</returns>
    public string ToJson() {
      return JsonConvert.SerializeObject(this, Formatting.Indented);
    }

}
}

File: generated-sources/dart/mojang-api/com/github/asyncmc/mojang/api/dart/model/security_challenge.dart
part of openapi.api;

class SecurityChallenge {
  
  SecurityQuestion question = null;
  
  SecurityAnswerId answer = null;
  SecurityChallenge();

  @override
  String toString() {
    return 'SecurityChallenge[question=$question, answer=$answer, ]';
  }

  SecurityChallenge.fromJson(Map<String, dynamic> json) {
    if (json == null) return;
    question = new SecurityQuestion.fromJson(json['question']);
    answer = new SecurityAnswerId.fromJson(json['answer']);
  }

  Map<String, dynamic> toJson() {
    return {
      'question': question,
      'answer': answer
    };
  }

  static List<SecurityChallenge> listFromJson(List<dynamic> json) {
    return json == null ? new List<SecurityChallenge>() : json.map((value) => new SecurityChallenge.fromJson(value)).toList();
  }

  static Map<String, SecurityChallenge> mapFromJson(Map<String, dynamic> json) {
    var map = new Map<String, SecurityChallenge>();
    if (json != null && json.length > 0) {
      json.forEach((String key, dynamic value) => map[key] = new SecurityChallenge.fromJson(value));
    }
    return map;
  }
}

